class.photon.php

```diff
@@ -720,7 +718,11 @@ public function filter_srcset_array( $sources, $size_array, $image_src, $image_m
 					);
 			} // foreach ( $multipliers as $multiplier )
 			if ( is_array( $newsources ) ) {
-				$sources = array_replace( $sources, $newsources );
+				if ( function_exists( 'array_replace' ) ) { // PHP 5.3+, preferred
+					$sources = array_replace( $sources, $newsources );
```

Comment:
Why don't we just use `array_replace_recursive()` directly?  Core polyfills it on 5.2, and 5.3+ should have it natively.  Should simplify some logic.

https://github.com/WordPress/WordPress/blob/dd6da701b286579819cd6aa518aa2d7018efd759/wp-includes/compat.php#L438-L489